woocommerce/woocommerce-android
# [Coupons] UI Design for Coupons List Screen

Part of #5527 

A bit of a long PR but some of it is just example data.



### Description


This implements the coupon details screen design below, with some details* and exception**:

<img width="520" alt="Screen Shot 2022-03-18 at 15 11 57" src="https://user-images.githubusercontent.com/266376/158962498-0fef414b-c434-480c-b1bb-1a8d12f19647.png">

__[*] Details__
Added items:
- Basic fragment, Compose components, and view model
- Navigation from "Coupons" button on More Menu to the Coupon List screen
- Couple of small string-related utilities.
- The design above is for the iOS version and contains a right-chevron. In Android it is not needed, so is not added.

__[**] Exception__
These are not added yet:
- Feedback form at the top (this is worked on #6067)
- Search and "+" icon on top right


### Testing instructions


1. Run app, go to more Menu, then tap the Coupons button
2. Check and compare the shown UI with the Figma design

### Images/gif



| Dark | Light |
|-|-|
| ![Dark version of Coupons List UI](https://user-images.githubusercontent.com/266376/159396553-1e2eccb3-9911-40e2-a6e0-8da6f34bbb4a.png) | ![Light version of Coupons List UI](https://user-images.githubusercontent.com/266376/159396542-e8383821-daf3-49a0-b9c4-1b0021fc4eeb.png) |
